src/nft/utils/txRoute/combineItemsWithTxRoute.ts

```diff
@@ -36,14 +70,24 @@ export const combineBuyItemsWithTxRoute = (
     const newPriceInfo = item.updatedPriceInfo ? item.updatedPriceInfo : item.priceInfo
 
     // if the price changed
-    if (
-      route &&
-      'priceInfo' in route.assetOut &&
-      isPriceDiff(newPriceInfo.basePrice, route.assetOut.priceInfo.basePrice)
-    ) {
-      return {
-        ...item,
-        updatedPriceInfo: route.assetOut.priceInfo,
+    if (route && 'priceInfo' in route.assetOut) {
```

Comment:
for this one, I think we can do some early returns, should help reduce the indentation @cbachmeier